portfolio/src/main/java/com/google/sps/data/BlogHashMap.java

```diff
@@ -122,7 +126,7 @@ public void putInMap(List<BlogMessage> messages) {
       return getMessages(tags.get(0), limit);
```

Comment:
Can we have a check in the very beginning like

if (!map.containsKey(tag)) {
  return getMessage(tags.get(0), limit);
}

to keep the main logic more focused?

BTW, you may need a comment for "tags.get(0)" to tell readers why using the first tag.